File: FBuild/FBuild.lua

if Build == nil then
   Build = "Release" 
end

if Build ~= "Release" and Build ~= "Debug" then 
   error("Build='<Release|Debug>' expected");
end;

FBuild.BuildExe({
                   Output = "../" .. Build .. "/FBuild.exe";
                   Config   = Build;
                   Outdir   = Build;
                   Includes = { "/boost_1_50_0" };
                   CRT      = 'Static';
                   Files    = FBuild.Glob("*.cpp");
                   PrecompiledHeader = {         
                      Header = 'Precompiled.h';  
                      Cpp    = 'Precompiled.cpp';
                   };
                   Libpath = { "/boost_1_50_0/stage/lib", "../" .. Build };
                   Libs = { "Lua.lib", "Shlwapi.lib" };
                });

File: FBuild/FBuild.js

if (args.build == undefined) args.build = "Release";
if (args.build != "Debug" && args.build != "Release") throw "build='<Release|Debug>' expected";

if (args.rebuild == undefined) args.rebuild = false;

ToolChain("x64");

var exe = new Exe;
exe.Build(args.build);
exe.Files(Glob("*.cpp"));
exe.DependencyCheck(!args.rebuild);
exe.CRT("Static");
exe.Includes("../Boost");
exe.Defines("_CRT_SECURE_NO_WARNINGS");
exe.PrecompiledHeader("Precompiled.h", "Precompiled.cpp");
exe.CompileArgs("-std:c++latest");
exe.WarningLevel(4).WarningAsError(true);

exe.Output("../" + args.build + "/FBuild.exe");
exe.LibPath("../Boost/stage/lib", "../" + args.build);
exe.Libs("Duktape.lib", "Ws2_32.lib", "Winmm.lib", "Shlwapi.lib");

exe.Create();
